src/trivia/util.h

```diff
@@ -675,6 +675,19 @@ thread_sleep(double sec);
 char *
 getenv_safe(const char *name, char *buf, size_t buf_size);
 
+/**
+ * Returns the string converted to lower case in-place.
+ */
+char *
+to_lower(char *s, size_t len);
```

Comment:
Please see [discussion](https://github.com/tarantool/tarantool/pull/7994#discussion_r1040850919).

Reply:
> Please see [discussion](https://github.com/tarantool/tarantool/pull/7994#discussion_r1040850919).

It doesn't address any of my points. Anyway, the boost library API isn't something we should use as an example.

Reply:
> I don't understand the purpose of to_lower_copy. One could call xstrdup + strtolower instead.

In your approach the string is iterate over thrice (strlen, strcpy, tolower) — in my approach it iterated over twice (strlen, tolower).

Reply:
As for me this is not so important thing. But if you ask - we should duplicate strings with `strdup` or `xstrdup`. I bet it duplicate a string in _almost_ one traverse :). I would also prefer if standard `tolower` is used only once, it would be more simple to support. Even more, `strtolower(xstrdup(..))` is the perfect solution: no additional function - no additional support cost.

As for performance - I bet that here the code with formally three scans will be at least not slower than the code with two. Practically I think that we could not make a benchmark that will show the difference.

Reply:
> As for performance - I bet that here the code with formally three scans will be at least not slower than the code with two. Practically I think that we could not make a benchmark that will show the difference.

Quick benchmark shows your assumption is wrong:
* Clang: https://quick-bench.com/q/HQZcsMiYmwThzoXpCu5XiewV0Ns
* GCC: https://quick-bench.com/q/c22Fwtx3QnEWgSCdVYXustEx54M

For some reason, Clang cannot effectively compile `strtolower(xstrdup(..))`.

Reply:
So, should I drop `strtolowerdup` despite the performance penatly?

Reply:
> Anyway, in the latest version of this PR, none of the new functions is ever used.

Now I use `strtolowerdup` for the key translation table.